src/loaders/genome_collection/parse_computed_genome_attribs.py

```diff
@@ -133,16 +192,366 @@ def _read_tool_result(result_dir, batch_dir, kbase_collection, load_ver, tool_fi
+
+        docs.extend(parse_ops(root_dir, kbase_collection, load_ver))
+
+    docs = merge_docs(docs, '_key')
+    output = f'{kbase_collection}_{load_ver}_{"_".join(genome_attr_tools)}_{COMPUTED_GENOME_ATTR_FILE_SUFFIX}'
+    _create_import_files(root_dir, output, docs)
+
+
+def _append_or_check_trait(
+        global_traits_meta: dict[str, dict[str, str]],
+        trait_id: str,
+        trait_index: int,
+        trait_name: str,
+        trait_description: str,
+        trait_category: str,
+        trait_type: str):
+    # Append a new trait to the global traits dictionary or check if the trait information is consistent
+
+    if trait_id not in global_traits_meta:
+        # add new trait
+        global_traits_meta[trait_id] = {
+            _SYS_TRAIT_INDEX: trait_index,
+            _SYS_TRAIT_NAME: trait_name,
+            _SYS_TRAIT_DESCRIPTION: trait_description,
+            _SYS_TRAIT_CATEGORY: trait_category,
+            _SYS_TRAIT_TYPE: trait_type,
+        }
+    else:
+        # check if the trait information is consistent
+        existing_trait = global_traits_meta[trait_id]
+        if (existing_trait[_SYS_TRAIT_NAME] != trait_name or
+                existing_trait[_SYS_TRAIT_INDEX] != trait_index or
+                existing_trait[_SYS_TRAIT_DESCRIPTION] != trait_description or
+                existing_trait[_SYS_TRAIT_CATEGORY] != trait_category or
+                existing_trait[_SYS_TRAIT_TYPE] != trait_type):
+            raise ValueError(f'Inconsistent trait information for trait {trait_id}')
+
+
+def _parse_categories(traits_meta: dict[str, dict[str, str]]) -> list[ColumnCategory]:
+    # Parse the trait categories from the global traits dictionary
+
+    categories = {}
+    # loop over each trait in traits_meta
+    for trait in traits_meta.values():
+        trait_category = trait[_SYS_TRAIT_CATEGORY]
+
+        if trait[_SYS_TRAIT_TYPE] == 'count':
+            trait_type = ColumnType.COUNT.value
+        elif trait[_SYS_TRAIT_TYPE] == 'binary':
+            trait_type = ColumnType.BOOL.value
+        else:
+            raise ValueError(f'Unknown trait type {trait[_SYS_TRAIT_TYPE]}')
+
+        # if the trait_category is not available, create a new ColumnCategory object with empty columns
+        if trait_category not in categories:
+            categories[trait_category] = ColumnCategory(category=trait_category, columns=[])
+
+        # add the trait to the appropriate category's list of columns
+        categories[trait_category].columns.append(ColumnInformation(id=str(trait[_SYS_TRAIT_INDEX]),
+                                                                    name=trait[_SYS_TRAIT_NAME],
+                                                                    description=trait[_SYS_TRAIT_DESCRIPTION],
+                                                                    type=trait_type))
+
+    # sort columns in each ColumnCategory object by the column id
```

Comment:
Remind me - for microtrait colum_id = microtrait-display-order?

Reply:
yes.

Reply:
:+1: